Applications/Spire/Source/UiViewer/TabTestWidget.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Spire/UiViewer/TabTestWidget.hpp"
+#include <QHBoxLayout>
+#include <QLabel>
+#include "Spire/Ui/TabWidget.hpp"
+
+using namespace Spire;
+
+TabTestWidget::TabTestWidget(QWidget* parent)
+    : QWidget(parent) {
+  auto layout = new QHBoxLayout(this);
+  auto tab_widget = new TabWidget(this);
+  layout->addWidget(tab_widget);
+  auto widget1 = new QLabel(tr("Tab 1"), this);
+  widget1->setAlignment(Qt::AlignCenter);
```

Comment:
Do your best to avoid using generic names like ```widget```, as someone reading the code, or even you in the future reading your own code, seeing a name like ```widget1```, and ``` widget2``` is unhelpful. At the bare minimum naming this "label1``` is immediately more descriptive than ```widget1```, then naming it ```tabLabel1``` is even better, it lets me know that this variable is being used to represent the label on a tab rather than just some arbitrary non-descript widget.

Now that you have nailed down the fundamentals in terms of good understanding of C++, take every measure you can to fine tune your code. Use good naming conventions, identify common patterns and factor them out, be very discriminating about every single line you commit and ensure that you are delivering the highest quality software you can.

Reply:
Is tab1_label acceptable? I vaguely recall numbers are required to come last, but that may have been a different style guideline.

Reply:
Yes you are correct, ```tab1_label```. It's a grammatical question not a coding style question; the number identifies the tab, not the label. ```tab_label_1``` means that there is a single tab with multiple labels whereas ```tab1_label``` means that there are multiple tabs and each tab has a label. Thanks for that correction.